### PR TITLE
Artillery impact tablet

### PR DESCRIPTION
**Pull request by breadhunt**
## About The Pull Request

Continuation of https://github.com/tgstation/TerraGov-Marine-Corps/pull/10555
image

The TGMC now uses orbital satellite technology to provide information for artillery operators on the quality of their shots. Push wipes, now caught in 4K Ultra HD.
Adds one of a special impact hud tablet to mortar and howitzer crates. The tablet displays a feed of any active artilley. When a shot is fired, the view changes to the impact site.

## Why It's Good For The Game

This has been wanted for a long time. Visual feedback good.
Gives visual feedback for loaders (which currently has 0 feedback beyond 'GOOD HITS') and potentially a way for spotters to check their hits.
This _isn't_ NVG. Loaders are not being powercrept into seeing in the dark across the map. They can, however, see if they hit 10 marines and obliterated the entire push (again). This **only** provides visual feedback for a loader - having a spotter is still way way way way better than the loader blindfiring.

## Changelog
:cl:
add: Mortar and howitzer crates now each have one artillery impact tablet, which can be used to get a live feed of artillery units and displays impact cams to observe hits
/:cl:

### CODE REVIEW

**Review by TiviPlus — CHANGES_REQUESTED** [review]
use return ..() and fix commit trash

**Comment by MereWalker** [comment]
This will greatly increase the usage of mortars and artillery. Look forward to it! Should make for some fun meme videos!

**Comment by Lumipharon** [comment]
this also allows for creative across the map scouting, not sure if that's desired or not.

**Comment by breadhunt** [comment]
> this also allows for creative across the map scouting, not sure if that's desired or not.

I considered that but I think that it's fine.Tadpole NVG does the same thing (AND MESONVISION FOR CAVES), for entirely *free* and to my knowledge I am the only person who's used it for scouting frequently. HE won't show anything (since it'll be pitch black) and you only get an additional second of window after impact to see what's going on. Additionally, tac binocs/rangefinders are both NVG - which already provides a ton of scouting potential.

TL;DR yes, but flare shells will have to be used for it. Marines already have better and more accessible util for entirely free already.

File: code/game/objects/machinery/mortar.dm [hunk 1]
```diff
@@ -63,6 +67,16 @@
 	///Used for remote targeting by AI
 	var/obj/item/ai_target_beacon/ai_targeter
 
+/obj/machinery/deployable/mortar/Initialize(mapload, _internal_item, deployer)
+	. = ..()
+	impact_cam.forceMove(src)
+	impact_cam.artillery = src
+
+/obj/machinery/deployable/mortar/Destroy()
+	qdel(impact_cam)
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:76]: QDEL_NULL

File: code/game/objects/machinery/mortar.dm [hunk 2]
```diff
@@ -258,6 +275,12 @@
 	say("Remote targeting set by [user]. COORDINATES: X:[coords["targ_x"]] Y:[coords["targ_y"]] OFFSET: X:[coords["dial_x"]] Y:[coords["dial_y"]]")
 	playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
 
+///Returns the impact camera to the mortar
+/obj/machinery/deployable/mortar/proc/return_cam()
+	current_shots--
+	if(current_shots == 0)
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:281]:
```suggestion
	if(current_shots <= 0)
```

File: code/game/objects/machinery/mortar.dm [hunk 3]
```diff
@@ -373,6 +396,26 @@
...
+	resistance_flags = RESIST_ALL
+	var/obj/machinery/deployable/mortar/artillery = null
+
+/obj/machinery/camera/artillery/update_icon()
+	icon_state = "nothing"
+
+/obj/machinery/camera/artillery/Destroy()
+	if(artillery)
+		artillery.impact_cam = null
+	return ..()
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:416]: i dont get why this subtype exists

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/mortar.dm (hunk 3 above)
Comment (breadhunt) [reply code/game/objects/machinery/mortar.dm:416]: -Regular cameras are visible, we want this one to be invisible
-Regular cameras are not invincible. If cameras aren't invincible, they get destroyed by the mortar explosion which ends the feed too early
-If the camera is deleted we need to clean the var in the mortar (I think)
-A unique c_tag makes this UI easier to distinghish from other camera consoles (but could be set on mortar init())
-Internal_light needs to be set to false, otherwise AI can magically light the camera up
-Network needs to be set to terragovartillery so we have a unique camera net for this (because we don't want colony camera consoles, etc. being able to tap into this)

if there's a cleaner way of handling it, that'd be a better idea, though!

Comment (TiviPlus) [reply code/game/objects/machinery/mortar.dm:416]: artillery var is completely unused
update_icon should not be overriden

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/mortar.dm [hunk 4]
```diff
@@ -45,6 +45,10 @@
 	var/reload_time = 0.5 SECONDS
 	///Amount of shells to fire if this is empty all shells in chamber items list will fire
 	var/fire_amount
+	///Camera to display impact shots
+	var/obj/machinery/camera/artillery/impact_cam = new
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:49]: put the new in initialize

File: code/game/objects/machinery/mortar.dm [hunk 5]
```diff
@@ -373,6 +395,16 @@
...
+// Artillery cameras. Together with the artillery impact hud tablet, shows a live feed of imapcts.
+
+/obj/machinery/camera/artillery
+	name = "artillery camera"
+	network = list("terragovartillery")
+	alpha = 0 //we shouldn't be able to see this!
+	internal_light = FALSE
+	c_tag = "impact camera"
+	resistance_flags = RESIST_ALL
+
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:407]: this seems like  a vary random place to define this shouldnt you put it above the mortar definition?

**Review by TiviPlus — CHANGES_REQUESTED** [review]
